linode/resource_linode_lke_cluster.go

```diff
@@ -406,6 +414,150 @@ func reconcileLKEClusterPoolSpecs(poolSpecs []linodeLKEClusterPoolSpec, pools []
 	return
 }
 
+func waitForClusterPoolReady(ctx context.Context, client *linodego.Client, errCh chan<- error, wg *sync.WaitGroup, clusterID, poolID int) {
+	eventTicker := time.NewTicker(500 * time.Millisecond)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Printf("[ERROR] timed out waiting for LKE Cluster (%d) Pool (%d) to be ready", clusterID, poolID)
+			return
+
+		case <-eventTicker.C:
+			pool, err := client.GetLKEClusterPool(ctx, clusterID, poolID)
+			if err != nil {
+				errCh <- fmt.Errorf("failed to get LKE Cluster (%d) Pool (%d): %w", clusterID, poolID, err)
+			}
+
+			ready := true
+			for _, instance := range pool.Linodes {
+				if instance.Status == linodego.LKELinodeNotReady {
+					ready = false
+				}
+			}
+
+			if ready {
+				log.Printf("[DEBUG] finished waiting for LKE Cluster (%d) Pool (%d) to be ready", clusterID, poolID)
+				wg.Done()
+				return
+			}
+		}
+	}
+}
+
+func waitForClusterPoolsToStartRecycle(ctx context.Context, client *linodego.Client, clusterID int, pools []linodego.LKEClusterPool) (<-chan int, <-chan error) {
+	clusterInstances := make(map[int]int)
+	poolInstances := make(map[int]map[int]struct{}, len(pools))
+	for _, pool := range pools {
+		poolInstances[pool.ID] = make(map[int]struct{}, len(pool.Linodes))
+		for _, instance := range pool.Linodes {
+			poolInstances[pool.ID][instance.InstanceID] = struct{}{}
+			clusterInstances[instance.InstanceID] = pool.ID
+		}
+	}
+
+	poolRecyclesCh := make(chan int)
+	errCh := make(chan error)
+
+	eventTicker := time.NewTicker(200 * time.Millisecond)
+	lastEventID := 0
+
+	go func() {
+		defer eventTicker.Stop()
+		defer close(poolRecyclesCh)
+		defer close(errCh)
+
+		for len(clusterInstances) != 0 {
+			select {
+			case <-ctx.Done():
+				log.Printf("[ERROR] timed out waiting for all original nodes of LKE Cluster (%d) to be deleted (%d remaining)\n", clusterID, len(clusterInstances))
+				return
+
+			case <-eventTicker.C:
+				filterBytes, _ := json.Marshal(map[string]interface{}{
+					"+order_by": "created",
```

Comment:
Is ID a valid value here? Makes me nervous that one logic is on dates, and the other is on IDs.